transaction/context.go

```diff
@@ -0,0 +1,45 @@
+package transaction
+
+import (
+	"context"
+	"fmt"
+)
+
+type contextKey int
+
+const (
+	trackedKey contextKey = iota
+	transactionIDKey
```

Comment:
I missed it was actually a `type key string`. I was missing that, because it was `string` and not `int` as usual.

So yes, in terms on correctness nothing is improved.

I still like the fact it solves imports problem w/o sacrificing anything, so I'd like to keep that. But I won't cry if you say no.